info.php

```diff
@@ -121,231 +52,318 @@ function check_if_wiris_button_are_in_atto_toolbar() {
     global $CFG;
 
-    $testname = get_string('lookingforwirisplugin', 'filter_wiris') . '&nbsp; TinyMCE';
-    $reporttext = get_string('wirispluginfor', 'filter_wiris') . '&nbsp; TinyMCE &nbsp;' .
-                    get_string('mustbeinstalled', 'filter_wiris');
-    $wirisplugin = '';
-    $condition = file_exists($wirisplugin);
-
-    if (!$condition) {
-        if ($CFG->branch < 402) {
-            $wirisplugin = '../../lib/editor/tinymce/plugins/tiny_mce_wiris';
-        } else {
-            $wirisplugin = '../../lib/editor/tiny/plugins/wiris';
-        }
-        $condition = file_exists($wirisplugin);
+    if ($editorname === 'tinymce' && $CFG->branch > 402 ) {
+        // if Moodle version is 4.1 or prior, do not check if tiny (legacy) exists
+        return null;
     }
 
-    return html_writer::tag('tr', wrs_createtablerow($testname, $reporttext, $solutionlink, $condition), array('class' => 'wrs_plugin wrs_filter'));
+    // get_texteditor returns (boolean)false if not exists or an object if the editor exists
+    $editor = get_texteditor($editorname);
+
+    if ($editor === false) {
+        return false;
+    } else {
+        return true;
+    }
 }
 
-function create_table_close() {
-    $output = '';
-    $output .= html_writer::end_tag('table');
+function get_exists_mt_editor($existeditor, $plugins, $editorname) {
+    if(!$existeditor) {
+        return null;
+    }
 
-    $output .= html_writer::start_tag('p');
-    $output .= html_writer::start_tag('br');
-    return $output;
+    return isset($plugins[$editorname]);
 }
 
-function warning_tiny_incompatibility() {
-    global $CFG;
-
-    if ($CFG->version < 2022112807) {
-        return;
+function get_mt_editor_version($exismtteditor, $plugins, $editorname) {
+    if(!$exismtteditor) {
+        return null;
     }
 
-    if (is_dir($CFG->dirroot.'/lib/editor/tinymce/plugins/tiny_mce_wiris')) {
-        \core\notification::warning(get_string('tinymceincompatibility', 'filter_wiris'));
-    }
+    return $plugins[$editorname]['release'];
 }
 
-// Page prologue.
-$PAGE->set_context(context_system::instance());
-$PAGE->set_title(get_string('title', 'filter_wiris'));
-$PAGE->set_url('/filter/wiris/info.php', array());
-echo $OUTPUT->header();
-
-// Get editor data for tests.
-$currenteditordata = get_current_editor_data($CFG->branch, $CFG->version, $CFG->texteditors);
+function get_mt_editor_enabled($existsmteditor, $editorname) {
+    if(!$existsmteditor){
+        return null;
+    }
 
-$solutionlink = 'https://docs.wiris.com/mathtype/en/mathtype-for-lms/mathtype-for-moodle.html#install-mathtype-for-moodle?utm_source=moodle&utm_medium=referral';
+    try {
+        $condition = check_if_wiris_button_are_in_toolbar($editorname);
+        return ($condition);
+    } catch (Exception $e) {
+        return null;
+    }
+}
 
-warning_tiny_incompatibility();
+function get_instalation_check($filterenabled, $filterversion, $enabledpluginsversion) {
```

Comment:
I'm missing maybe a one line explanation on the row check this method is for. Like "Validates that MathType is usable at least in one of the text editors supported" or something you see better fit!